CONTRIBUTING.md

```diff
@@ -53,3 +53,92 @@ To build a local image use the `bin/task docker-local` command in the [FerretDB]
+Diff tests are located in the `tests/diff` directory and are regular Go tests.
+
+Currently, every diff test or subtest should have at least two subtests - `FerretDB` and `MongoDB`.
+Those subtests usually only contain asserts based on the expected behavior of each database.
+
+Let's take a look at the following example:
+
+```go
+func TestDollarSign(t *testing.T) {
+    t.Run("Insert", func(t *testing.T) {
+        _, err := db.Collection("collection").InsertOne(ctx,  bson.D{{"foo$", "bar"}})
+
+        t.Run("FerretDB", func(t *testing.T) {
+            expected := mongo.CommandError{
+                Code:    2,
+                Name:    "BadValue",
+                Message: `invalid key: "foo$" (key must not contain '$' sign)`,
+            }
+            AssertEqualError(t, expected, err)
+        })
+
+        t.Run("MongoDB", func(t *testing.T) {
+            require.NoError(t, err)
+        })
+    })
+
+    /* Further subtests */
+}
+```
+
+It works with document validation feature for the `foo$` field key
+and demonstrates how FerretDB validation differs from MongoDB behavior.
+In case of FerretDB a document is considered  invalid if it has a key containing the `$` sign.
+In case of MongoDB such document is considered valid.
+The subtest `Insert` demonstrates this difference: in case of FerretDB we expect to receive a particular error,
+in case of MongoDB we expect to receive no error when we insert such document.
+
+### Configuration
+
+Configuration of diff tests is stored in the `tests/diff.yml` file.
+In particular, this file contains the information
+about expected number of failed and passed tests for each database and what is considered a failed and a passed test.
+
+With the current configuration, we expect that for FerretDB all the subtests that match regular expression `FerretDB$` pass
+and all the subtests that don't match this regular expression fail.
+So, for FerretDB the number of passed tests is equal to the number
+of all the `FerretDB` subtests running.
+The number of failed tests is calculated hierarchically: the "parental" test
+and all its subtests that don't match the regular expression `FerretDB$` are considered failed.
+For MongoDB, we have a similar configuration.
+
+In the example above (`TestDollarSign`), for FerretDB, the number of passed tests is 1 (for `FerretDB` subtest).
+The number of failed tests is 3 (`MongoDB` subtest, `Insert`, and finally `TestDollarSign`).
+
+### How it works?
+
+To run diff tests call `task dance DB=ferretdb TEST=diff` command to test FerretDB
+or `task dance DB=mongodb TEST=diff` to test MongoDB.
+
+The `dance` command reads the configuration file and runs the tests.
+Then it compares
+the actual number of passed and failed tests with the number from the configuration file.
+
+If the numbers are different, dance prints the list of unexpected tests and exits with a non-zero code.
+
+### Developer workflow
+
+A typical developer workflow when working with diff tests is as follows:
```

Comment:
The typical developer workflow does not include diff tests at all. And again, we should not duplicate information

Reply:
"A typical developer workflow when working with diff tests" does not include diff tests?

Reply:
Did we add diff tests during this typical sprint?

Reply:
Well, I meant "typical developer workflow when working with diff tests", not "typical developer workflow" in general.  How to rephrase it to highlight that it's only when such tests are included?

Reply:
fixed